haystack/utils/openai_utils.py

```diff
@@ -64,46 +64,40 @@ def _openai_text_completion_tokenization_details(model_name: str):
 
     :param model_name: Name of the OpenAI model.
     """
-    tokenizer_name = "gpt2"
-    max_tokens_limit = 4096  # It is the minimum max_tokens_limit value based on this ref: https://platform.openai.com/docs/models/overview
+    tokenizer_name = "cl100k_base"
+    # It is the minimum max_tokens_limit value based on this ref: https://platform.openai.com/docs/models/overview
+    max_tokens_limit = 4096
     try:
         model_tokenizer = tiktoken.encoding_name_for_model(model_name)
     except KeyError:
         model_tokenizer = None
 
     if model_tokenizer:
-        # Based on OpenAI models page, the following are the max_tokens_limit values for the corresponding models
-        ## Ref: https://platform.openai.com/docs/models/overview
-        if "davinci" in model_name:
+        tokenizer_name = model_tokenizer
+        if model_name == "davinci-002" or model_name == "babbage-002":
             max_tokens_limit = 16384
-            tokenizer_name = model_tokenizer
-        elif (
-            model_name.startswith("gpt-3.5-turbo-instruct")
-            or model_name.startswith("gpt-35-turbo-instruct")
-            or model_name.startswith("gpt-3.5-turbo-0613")
-            or model_name.startswith("gpt-35-turbo-0613")
+
+        # GPT-3.5 models that have a different token limit than 4096
+        # Handles default case for GPT-3.5 models
+        if model_name.startswith("gpt-3.5-turbo") or model_name.startswith("gpt-35-turbo"):
+            max_tokens_limit = 16385
+        # Handles two edge-cases where the value is 4096
+        if (
+            model_name == "gpt-3.5-turbo-instruct"
+            or model_name == "gpt-3.5-turbo-0613"
+            or model_name == "gpt-35-turbo-instruct"
+            or model_name == "gpt-35-turbo-0613"
         ):
             max_tokens_limit = 4096
```

Comment:
I would indent this block:
looks like a subcase of the previous condition

Reply:
Sounds good I did the same with the gpt-4 block as well